app/models/setting.rb

```diff
@@ -175,6 +175,14 @@ def self.create!(opts)
     end
   end
 
+  def self.regexp_expand_wildcard_string(string)
+    "\\A#{Regexp.escape(string).gsub('\*', '.*')}\\Z"
+  end
+
+  def self.convert_array_to_regexp(array)
+    Regexp.new(array.map {|string| regexp_expand_wildcard_string(string) }.join('|'))
```

Comment:
This errors out when settings return nil (not sure how I got there, I do see an array in the menu). Anyway, can you add ` || []` in the `fact_parser.rb` there?

```
2015-10-08T08:56:34 [app] [D] undefined method `map' for nil:NilClass
 | /home/lzap/work/foreman/app/models/setting.rb:183:in `convert_array_to_regexp'
 | /home/lzap/work/foreman/app/services/fact_parser.rb:166:in `ignored_interfaces'
 | /home/lzap/work/foreman/app/services/fact_parser.rb:170:in `block in remove_ignored'
 | /home/lzap/work/foreman/app/services/fact_parser.rb:170:in `delete_if'
 | /home/lzap/work/foreman/app/services/fact_parser.rb:170:in `remove_ignored'
 | /home/lzap/work/foreman/app/services/fact_parser.rb:66:in `interfaces'
 | /home/lzap/work/foreman/app/services/puppet_fact_parser.rb:90:in `interfaces'
 | /home/lzap/work/foreman_discovery/app/lib/puppet_fact_parser_extensions.rb:19:in `suggested_primary_interface_with_bootif'
 | /home/lzap/work/foreman/app/models/host/base.rb:165:in `set_interfaces'
 | /home/lzap/work/foreman/app/models/host/base.rb:148:in `populate_fields_from_facts'
 | /home/lzap/work/foreman_discovery/app/models/host/discovered.rb:75:in `populate_fields_from_facts'
 | /home/lzap/work/foreman/app/models/host/base.rb:124:in `import_facts'
 | /home/lzap/work/foreman_discovery/app/models/host/discovered.rb:60:in `import_facts'
 | /home/lzap/work/foreman_discovery/app/models/host/discovered.rb:52:in `import_host_and_facts'
```